src/keep_gpu/mcp/server.py

```
@@ -780,35 +792,37 @@ def _reject_session_route_components(self, parsed) -> bool:
-            return
 
-        if path.startswith("/api/"):
-            self._json_response(404, {"error": {"message": "Unknown endpoint"}})
-            return
+            if path.startswith("/api/"):
+                self._json_response(404, {"error": {"message": "Unknown endpoint"}})
+                return
 
-        self._serve_static(path)
+            self._serve_static(path)
+        except Exception as exc:  # pragma: no cover - defensive
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)